sinks/splunk/splunk.go

```diff
@@ -46,24 +54,75 @@ func NewSplunkSpanSink(server string, token string, localHostname string, valida
 		client.SetHTTPClient(httpC)
 	}
 
-	return &splunkSpanSink{Client: client, hostname: localHostname, log: log, sendTimeout: sendTimeout}, nil
+	return &splunkSpanSink{
+		Client:          client,
+		ingest:          make(chan *hec.Event),
+		flush:           make(chan []*hec.Event),
+		hostname:        localHostname,
+		log:             log,
+		sendTimeout:     sendTimeout,
+		maxSpanCapacity: maxSpanCapacity,
+	}, nil
 }
 
 func (sss *splunkSpanSink) Start(cl *trace.Client) error {
 	sss.traceClient = cl
+	go sss.batchAndSend()
+
 	return nil
 }
 
+func (sss *splunkSpanSink) batchAndSend() {
+	batch := make([]*hec.Event, 0, sss.maxSpanCapacity)
+	for {
+		select {
+		case ev := <-sss.ingest:
+			batch = append(batch, ev)
```

Comment:
This doesn't ensure that the array won't exceed `maxSpanCapacity`. Once the capacity is exceeded, every append operation will allocate the entire array all over again, which I don't think is what we want.

Reply:
Correct. The check happens at `Ingest`, using [that atomic load](https://github.com/stripe/veneur/pull/538/files#diff-e55a3e6f32b35a56b86512a7390d0dbbR146).